src/main/java/org/apache/sling/auth/oauth_client/ClientConnection.java

```diff
@@ -34,4 +35,10 @@ public interface ClientConnection {
      * @return the name of the connection
      */
     @NotNull String name();
+    String clientId();
```

Comment:
SonarQube checks are not mandatory, only Jenkins is.

I would prefer to keep the duplication and not expose this API to consumers.